cloud/amazon/ec2_ami.py

```diff
@@ -225,6 +225,12 @@ def create_image(module, ec2):
                   'description': description,
                   'no_reboot': no_reboot}
 
+        images = ec2.get_all_images(filters={'name': name})
+
+        if images and images[0]:
+            module.exit_json(msg="AMI name already present", image_id=images[0].id, state=images[0].state, changed=False)
+            sys.exit(0)
```

Comment:
sys.exit was in the original code, I just moved it. What should go in it's place?

I don't believe no change would be correct, as your playbook would continue on, but the AMI would not accurately reflect a machine in this scenarion:

Machine A
Machine B
Create AMI FOOBAR from A
Create AMI FOOBAR from B
Launch Machine C from FOOBAR

In this case FOOBAR would be from A, not B, which would be expected.